ci/test_python.sh

```diff
@@ -123,5 +107,57 @@ pytest \
   tests
 popd
 
+if [[ "${RAPIDS_CUDA_VERSION}" == "11.8.0" ]]; then
+  if [[ "${RUNNER_ARCH}" != "ARM64" ]]; then
+    rapids-mamba-retry env create --force -f env.yaml -n test_cugraph_pyg
+
+    # Temporarily allow unbound variables for conda activation.
+    set +u
+    conda activate test_cugraph_pyg
+    set -u
+    
+    # Install pytorch
+    rapids-mamba-retry install \
+      --force-reinstall \
+      --channel pytorch \
+      --channel pytorch-nightly \
+      --channel nvidia \
+      'pytorch>=2.0' \
+      'pytorch-cuda>=11.8'
+
+    # Install the nightly version of pyg
+    pip install pyg-nightly
+
+    rapids-print-env
+
+    rapids-mamba-retry install \
+      --channel "${CPP_CHANNEL}" \
+      --channel "${PYTHON_CHANNEL}" \
+      libcugraph \
+      pylibcugraph \
+      cugraph \
+      cugraph-pyg
+
```

Comment:
Two quick notes:

- `rapids-print-env` does a `conda list` ([src](https://github.com/rapidsai/gha-tools/blob/main/tools/rapids-print-env)). so it might be worth placing that command after all the mamba install commands.
- I notice that CI is installing the wrong `cugraph-pyg` version (see [here](https://github.com/rapidsai/cugraph/actions/runs/4486137081/jobs/7899136258?pr=3288#step:7:2410)). This is probably from a dependency issue somewhere. It might take a bit of iterating to figure out why this is happening. I wrote some docs about reproducing CI locally [here](https://docs.rapids.ai/resources/reproducing-ci/) that might help. Happy to hop on a call to help or discuss this more if needed.